terra_notebook_utils/terra_auth.py

```diff
@@ -0,0 +1,70 @@
+"""
+Support for auth with Terra backend services.
+"""
+from azure.core.exceptions import ClientAuthenticationError
+from google.auth.exceptions import DefaultCredentialsError
+
+from terra_notebook_utils import azure_auth, gs, ExecutionPlatform
+from terra_notebook_utils.logger import logger
+from terra_notebook_utils.utils import get_execution_context
+
+
+class AuthenticationError(Exception):
+    pass
+
+
+class TerraAuthTokenProvider:
+    """
+    Provides auth bearer tokens suitable for use with Terra backend services.
+    """
+    def __init__(self):
+        self.execution_context = get_execution_context()
+
+    @staticmethod
+    def _identify_valid_access_token() -> str:
+        """
+        Try to obtain an auth bearer tokens suitable for use with Terra backend services
```

Comment:
Nit: `an auth bearer tokens`

Reply:
Corrected, thanks for the careful review!